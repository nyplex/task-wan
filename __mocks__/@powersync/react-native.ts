module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
