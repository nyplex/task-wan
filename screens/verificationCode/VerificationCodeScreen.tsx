import { Box } from "@/components/gluestack/box";
import Header from "./components/Header";
import BodyContent from "./components/BodyContent";
import CodeForm from "./components/CodeForm";

const VerificationCodeScreen = () => {
  return (
    <Box className="flex-1 bg-backrgound py-safe-offset-4 px-4">
      <Header />
      <BodyContent />
      <CodeForm />
    </Box>
  );
};

export default VerificationCodeScreen;
