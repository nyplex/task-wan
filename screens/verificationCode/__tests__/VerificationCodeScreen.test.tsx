import { render } from "@testing-library/react-native";
import VerificationCodeScreen from "../VerificationCodeScreen";
import React from "react";

// Do NOT mock gluestack components as requested
// Mock only child components to isolate VerificationCodeScreen
const Header = () => <></>;
Header.displayName = "Header";
const BodyContent = () => <></>;
BodyContent.displayName = "BodyContent";
const CodeForm = () => <></>;
CodeForm.displayName = "CodeForm";
const ResendCodeBtn = () => <></>;
ResendCodeBtn.displayName = "ResendCodeBtn";

jest.mock("../components/Header", () => ({ __esModule: true, default: Header }));
jest.mock("../components/BodyContent", () => ({ __esModule: true, default: BodyContent }));
jest.mock("../components/CodeForm", () => ({ __esModule: true, default: CodeForm }));
jest.mock("../components/ResendCodeBtn", () => ({ __esModule: true, default: ResendCodeBtn }));

describe("VerificationCodeScreen", () => {
  it("renders without crashing", () => {
    render(<VerificationCodeScreen />);
  });

  it("renders Header, BodyContent, CodeForm, and ResendCodeBtn", () => {
    const { UNSAFE_queryAllByType } = render(<VerificationCodeScreen />);
    expect(UNSAFE_queryAllByType(Header).length).toBe(1);
    expect(UNSAFE_queryAllByType(BodyContent).length).toBe(1);
    expect(UNSAFE_queryAllByType(CodeForm).length).toBe(1);
    expect(UNSAFE_queryAllByType(ResendCodeBtn).length).toBe(1);
  });

  it("matches the rendered output snapshot", () => {
    const { toJSON } = render(<VerificationCodeScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
