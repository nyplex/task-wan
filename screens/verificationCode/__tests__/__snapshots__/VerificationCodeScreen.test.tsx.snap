// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationCodeScreen matches the rendered output snapshot 1`] = `
<View
  className="flex-1 bg-backrgound py-safe-offset-4 px-4"
>
  <View
    className="mt-12"
  />
</View>
`;
